### PR TITLE
Support terminating process running test via its test invocation

### PR DESCRIPTION
**Pull request by happz**
Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

File: tmt/steps/execute/__init__.py [hunk 1]
```diff
@@ -271,6 +274,35 @@ def handle_reboot(self) -> bool:
...
 
+    def terminate_process(
+            self,
+            signal: _signal.Signals = _signal.SIGTERM,
+            logger: Optional[tmt.log.Logger] = None) -> None:
+        """
+        Terminate the invocation process.
+        .. warning::
+            This method should be used carefully. Process running the
+            invocation's test has been started by by some part of tmt code which
```
Comment (janhavlin) [inline tmt/steps/execute/__init__.py:285]:
```suggestion
            invocation's test has been started by some part of tmt code which
```

**Review by thrix — CHANGES_REQUESTED** [review]
@happz a unit test would be nice, should not be hard to test?

**Comment by happz** [comment]
> @happz a unit test would be nice, should not be hard to test?

Added in a36cc1c80cc117964b0b0dc470a5f6103690ea23

File: tmt/steps/execute/__init__.py (hunk 1 above)
Comment (happz) [reply tmt/steps/execute/__init__.py]: Fixed in 33bfbfb9

**Review by thrix — APPROVED** [review]
LGTM

**Review by psss — APPROVED** [review]
Looks good. Thanks.

**Comment by psss** [comment]
Ok, so now the `/plans/install/docs` was failing because of a real failure:

    docstring of tmt.steps.execute.TestInvocation.terminate_process:3: ERROR: Unexpected indentation.
    docstring of tmt.steps.execute.TestInvocation.terminate_process:8: WARNING: Literal block ends without a blank line; unexpected unindent.

Should be fixed by aa72ed56.

**Pull request merged**